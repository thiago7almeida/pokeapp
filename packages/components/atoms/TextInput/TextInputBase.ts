import styled from 'styled-components/native';
import {responsive} from '../../utils';

export const TextInputBase = styled.TextInput`
  height: ${responsive.size(40)}px;
`;
