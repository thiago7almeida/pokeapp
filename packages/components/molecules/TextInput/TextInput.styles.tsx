import styled from 'styled-components/native';
import {spacing} from '../../layouts';

export const TextInputContainer = styled.View`
  border: 1px solid;
  border-color: gray;
  border-radius: ${spacing.s32}px;
  padding-horizontal: ${spacing.s16}px;
  margin: ${spacing.s8}px;
`;
