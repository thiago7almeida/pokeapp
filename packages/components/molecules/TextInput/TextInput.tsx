import React from 'react';
import {TextInputProps} from 'react-native';

import {TextInputBase} from '@components';

import {TextInputContainer} from './TextInput.styles';

export const TextInput: React.FC<TextInputProps> = props => {
  return (
    <TextInputContainer>
      <TextInputBase autoCorrect={false} {...props} />
    </TextInputContainer>
  );
};
