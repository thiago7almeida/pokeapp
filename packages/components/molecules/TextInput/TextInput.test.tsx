import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {TextInput} from './TextInput';

describe('TextInput', () => {
  const onChangeText = jest.fn();
  const value = 'input value';
  const placeholder = 'placeholder text';

  it('should render correctly', () => {
    const {toJSON} = render(
      <TextInput
        value={value}
        placeholder={placeholder}
        onChangeText={onChangeText}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should pass props to TextInputBase', () => {
    const {getByPlaceholderText} = render(
      <TextInput
        value={value}
        placeholder={placeholder}
        onChangeText={onChangeText}
      />,
    );
    const input = getByPlaceholderText(placeholder);

    expect(input.props.value).toEqual(value);
    expect(input.props.onChangeText).toEqual(onChangeText);
  });

  it('should disable autocorrect', () => {
    const {getByPlaceholderText} = render(
      <TextInput
        value={value}
        placeholder={placeholder}
        onChangeText={onChangeText}
      />,
    );
    const input = getByPlaceholderText(placeholder);

    expect(input.props.autoCorrect).toBe(false);
  });

  it('should call onChangeText when typing', () => {
    const {getByPlaceholderText} = render(
      <TextInput
        value={value}
        placeholder={placeholder}
        onChangeText={onChangeText}
      />,
    );
    const input = getByPlaceholderText(placeholder);

    fireEvent.changeText(input, 'new value');

    expect(onChangeText).toHaveBeenCalledWith('new value');
  });
});
