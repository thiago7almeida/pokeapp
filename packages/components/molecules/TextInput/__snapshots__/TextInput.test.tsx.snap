// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<View
  style={
    [
      {
        "borderColor": "gray",
        "borderRadius": 59,
        "borderStyle": "solid",
        "borderWidth": 1,
        "marginBottom": 15,
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 15,
        "paddingHorizontal": 29,
      },
    ]
  }
>
  <TextInput
    autoCorrect={false}
    onChangeText={[MockFunction]}
    placeholder="placeholder text"
    style={
      [
        {
          "height": 73,
        },
      ]
    }
    value="input value"
  />
</View>
`;
